README.rst

```diff
@@ -0,0 +1,82 @@
+######
+ts_CBP
+######
+
+CBP is a Commandable SAL Component for the `Vera C. Rubin Observatory <https://lsst.org>`_.
+It controls a Collimated Beam Projector for the Main Telescope Calibration System.
+
+Installation
+============
+
+pip/setuptools method
+
+.. code::
+
+    pip install -e .[dev]
+    pytest --cov lsst.ts.cbp -ra
+
+EUPs method
+
+.. code::
+
+    pip install .[dev]
+    setup -kr .
+    scons 
+
+Requirements
+------------
+This project uses the ``black`` linter.
+
+.. code::
+
+    pip install .[dev]
+    pre-commit install
+
+``black`` will now run as a pre-commit git hook.
```

Comment:
I think you have to enable it? Certainly you do if using .githooks, as per ts_salobj's README:

```
This code is automatically formatted by ``black`` using a git pre-commit hook.
To enable this:

* Install the ``black`` Python package.
* Run ``git config core.hooksPath .githooks`` once in this repository.
```

Reply:
`pre-commit install` installs the git hook for the developer. This runs `black` before the work is committed and then the changes have to be added before the commit completes sucessfully. This is according to the official `black` installation guide. The Github action will catch issues if black has not been run and fail the build.

Reply:
The `pre-commit` package installs a huge number of dependencies and is overkill. You can replace all that with the following trivial file: `.githooks/pre-commit` which does the same thing:
```
#!/bin/sh
black . --check
```
The user has to run `git config core.hooksPath .githooks` once (in the repo's directory) to enable the hook.

It is easy to also do the formatting for the user (though unfortunately it is not possible to then allow the commit to proceed -- that results in the incorrectly formatted code being committed, not the formatted code, alas).

The following pre-commit file does that, and is what I plan to use from now on:
```
#!/bin/sh
if black . --check; then
    echo "Your code is black formatted."
else
    echo "Reformatting your code with black; please try the commit again."
    black .
    exit 1
fi
```
My original file did the reformatting and allowed the (unformatted) commit to proceed, which was a nuisance, so I appreciate your pushing on this as it allowed me to improve things a lot. But I am still very reluctant to add `pre-commit` package to T&S software without a demonstrated need. Yes it is what black's documentation recommends, but this other technique works, and in a less magic way.